mslib/msui/constants.py

```diff
@@ -30,12 +30,17 @@
 import fs
 import os
 import logging
+import platformdirs
 
 # ToDo refactor to generic functions, keep only constants
 HOME = os.path.expanduser(f"~{os.path.sep}")
 MSUI_CONFIG_PATH = os.getenv("MSUI_CONFIG_PATH", os.path.join(HOME, ".config", "msui"))
 # Make sure that MSUI_CONFIG_PATH exists
-_ = fs.open_fs(MSUI_CONFIG_PATH, create=True)
+_fs = fs.open_fs(MSUI_CONFIG_PATH, create=True)
+# MSUI does not actually support any PyFilesystem2 fs that is not available as a local path
+MSUI_CONFIG_SYSPATH = _fs.getsyspath("")
+
+MSUI_CACHE_PATH = platformdirs.user_cache_path("msui", "mss")
 
```

Comment:
On the tutorials I have to remove these data too. There it would be easier to remove only the MSUI_CONFIG_PATH, which I can do by using a config, which I also clear.

Because it wasn't the same dir on a new tutorial I didn't care. 

It might easier to have a cache dir below the MSUI_CONFIG_PATH  so all is on one place.

Reply:
Also before we run tests the platformdirs cache needs to be cleared.

Reply:
> It might easier to have a cache dir below the MSUI_CONFIG_PATH so all is on one place.

Cache files don't belong into the configuration directory. There is a reason those are separate XDG directories. As a user I should be able to remove `XDG_CACHE_HOME` without any issue and be confident that this has cleared all application caches.

Reply:
> Also before we run tests the platformdirs cache needs to be cleared, or stored randomized.

That happens by setting XDG_CACHE_HOME to a temporary directory in `tests.constants`.

Reply:
on my KDE that is empty. 

![image](https://github.com/Open-MSS/MSS/assets/278728/0707b7cd-6b3c-4f20-aa14-ef142a9c963a)

Reply:
> On the tutorials I have to remove these data too. There it would be easier to remove only the MSUI_CONFIG_PATH, which I can do by using a config, which I also clear.

The cache should behave in a way that it doesn't matter if it exists already or not (just speeding things up if it does), so there should not necessarily be a need to remove it. Nonetheless, the tutorials could also just set XDG_CACHE_HOME (or a MSUI_CACHE_PATH, if we want to additionally introduce that).

> Because it wasn't the same dir on a new tutorial I didn't care.

It still left trash in /tmp though.

Reply:
> on my KDE that is empty.
> 
> ![image](https://private-user-images.githubusercontent.com/278728/331257588-0707b7cd-6b3c-4f20-aa14-ef142a9c963a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTU4NzE2ODUsIm5iZiI6MTcxNTg3MTM4NSwicGF0aCI6Ii8yNzg3MjgvMzMxMjU3NTg4LTA3MDdiN2NkLTZiM2MtNGYyMC1hYTE0LWVmMTQyYTljOTYzYS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNTE2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDUxNlQxNDU2MjVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0wYjk4MjQ4NjQzNGU5YjFkNDdhYWIxMTJkMjRlNWQ3YzE5NjgyZDAxZGY4OWViOTJhMmJkZmY5ZDMwMDFiZjE2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.ZBAAcAWf3Gx2EEKAPV5m62i3eNYMiB57PSsSxrPFiW8)

https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html:

>  $XDG_CACHE_HOME defines the base directory relative to which user-specific non-essential data files should be stored. If $XDG_CACHE_HOME is either not set or empty, a default equal to $HOME/.cache should be used.

Reply:
Will see next days when I  recreate the tutorials if we have to adopt something.